module/os/linux/spl/spl-proc.c

```diff
@@ -648,21 +651,36 @@ static struct ctl_table spl_dir[] = {
 
 static struct ctl_table spl_root[] = {
 	{
-	.procname = "kernel",
-	.mode = 0555,
-	.child = spl_dir,
+		.procname	= "kernel",
+		.mode		= 0555,
+		.child		= spl_dir,
 	},
 	{}
 };
+#endif
 
 int
 spl_proc_init(void)
 {
 	int rc = 0;
 
-	spl_header = register_sysctl_table(spl_root);
+#ifdef HAVE_REGISTER_SYSCTL_TABLE
+	spl_header = register_sysctl_table(spl_table);
```

Comment:
Shouldn't this be `spl_root` instead of `spl_table` to match the original code.

Better yet, I see the `register_sysctl()` interface go all the way back to the 3.4 kernel.  Why don't we simply check for the "new" interface and always use it or return an error at configure time with `ZFS_LINUX_TEST_ERROR`.  We only care about support back to the Linux 3.10 kernel.

Reply:
Yep, you're correct. That's a typo on my end.

Reply:
The latest code corrects this type